pygfx/renderers/wgpu/engine/renderer.py

```diff
@@ -489,9 +489,9 @@ def render(
 
         # Update transform uniform buffers
         for wobject in wobject_list:
-            transform = WorldObject.transform_updates.pop(wobject, None)
-            if transform:
-                wobject._update_uniform_buffers(transform)
+            if wobject._uniform_buffers_dirty:
+                wobject._update_uniform_buffers()
+                wobject._uniform_buffers_dirty = False
```

Comment:
What would the performance be if `transform_updates` would be a `WeakSet` of world objects, and this'd be:
```py
for wobject in WorldObject.transform_updates:
    wobject._update_uniform_buffers()
```

A disadvantage is that it updates all dirty transforms, not only of the wobjects that are going to be rendered just now. But an advantage is that you don't have to iterate over all wobjects.

Reply:
Well, that's almost exactly what it was! This is faster.

Reply:
I was triggered by this because I'm thinking how to improve performance for may objects, renderbundles being one solution. But we can revisit this when the time comes. There may be better ways to handle this "caching". So let's just stick with what you have now.

Reply:
Yeah, I understand, but this loop is not even visible in the profiling data. So it's just not worth improving further right now. flag_update is like 40% of the total runtime.